src/status_im2/contexts/onboarding/common/carousel/animation.cljs

```diff
@@ -0,0 +1,50 @@
+(ns status-im2.contexts.onboarding.common.carousel.animation
+  (:require
+    [react-native.reanimated :as reanimated]
+    [utils.worklets.onboarding-carousel :as worklets.onboarding-carousel]))
+
+(def progress (atom nil))
```

Comment:
I wonder is it possible to avoid these global atoms and instead have them scoped to the onboarding stack? 🤔

Reply:
Maybe, we can initialize them on top of stack and pass as params.
Although we have two roots, we have to do this initialization in profiles & intro.
But just curious why we want to do that?

Reply:
> we can initialize them on top of root and pass as params to other stacks

It sounds complicated to pass shared values as params to navigated screens

Reply:
Yeah I believe we want to avoid global atoms as much as possible
 
I understand there are exceptions and perhaps it's needed/ not so much of an issue in this case

cc: @ilmotta @ulisesmac

Reply:
Thanks for cc'ing me @J-Son89. Let's be real, global state is often not a big issue since most of our global atoms or ratoms are used only in the scope of a single namespace, at least that's how I feel. Practically speaking, it's passable :man_shrugging: 

But having said that, I still think it's weird... In one hand Clojure(Script) promotes pure code as much as possible and controlled, local state. On the other hand, ClojureScript and Reagent allow us to define state used for views outside view components, something that's a no-no in React itself. That is to say, even raw React code feels more functional than our own ClojureScript code with global vars.

It's a discussion without a clear answer to me. As usual, almost all global state is created in the name of convenience, so this is not enough to justify, at least to me.

Given this discussion come and go every now and then, maybe it's about time we write something in the guidelines if we can all agree on something. Perhaps the discussion should be made outside this PR?

Reply:
Great point, reusability is one big reason to favor local state, because there's no state to synchronize between components in the first place.